### PR TITLE
Update test fixtures

### PR DESCRIPTION
**Pull request by zzat**
- [x] #3
- [x]  #2

### CODE REVIEW

File: swift-ticketing/src/swift_ticketing/components/http_server.clj [hunk 1]
```diff
@@ -12,13 +12,14 @@
...
   (stop [component]
     (println ";; Stopping API Server")
     (.stop (:http-server component))
     (assoc component :http-server nil)))
 
-(defn new-http-server [port]
-  (map->HTTPServer {:port port}))
+(defn new-http-server [port join?]
+  (map->HTTPServer {:port port
+                    :join? join?}))
```
Comment (olttwa) [inline swift-ticketing/src/swift_ticketing/components/http_server.clj:25]: I couldn't find documentation about :join? parameter but asking chatgpt gave me this answer.

> The :join? parameter determines whether this line will block or not.

Is this correct?

File: swift-ticketing/test/swift_ticketing/fixtures.clj [hunk 2]
```diff
@@ -1,20 +1,53 @@
...
-  (let [db-config (:database (read-test-config))]
-    {:db-spec (create-connection-pool db-config)}))
+  {:test-user-id nil
+   :db-spec nil})
+
+(def swift-ticketing-test-system
+  (swift-ticketing-system (read-test-config)))
+
+(defn start-test-system []
+  (alter-var-root #'swift-ticketing-test-system component/start))
```
Comment (olttwa) [inline swift-ticketing/test/swift_ticketing/fixtures.clj:19]: @neenaoffline is this the right way to start a component process? Seems very unclojury to me 😱

File: swift-ticketing/test/swift_ticketing/fixtures.clj [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
 
-(defn run-migrations [tests]
-  (let [db-config (:database (read-test-config))]
+(defn stop-test-system []
+  (alter-var-root #'swift-ticketing-test-system component/stop))
+
+(defn setup-test-system [tests]
```
Comment (olttwa) [inline swift-ticketing/test/swift_ticketing/fixtures.clj:24]: This sets up a LOT of things. Not sure this should be called via a one-time fixture from handlers-test, but rather something more closer to running tests. Something like Goose's [test-runner](https://github.com/nilenso/goose/blob/main/test/test_runner.clj) maybe?

Comment (olttwa) [reply swift-ticketing/test/swift_ticketing/fixtures.clj:24]: Reason for pulling this part above is so when you have other tests that require DB setup, you don't have to do that via fixtures.

File: swift-ticketing/src/swift_ticketing/components/http_server.clj (hunk 1 above)
Comment (zzat) [reply swift-ticketing/src/swift_ticketing/components/http_server.clj:25]: Yes. You can find the documentation here: https://ring-clojure.github.io/ring/ring.adapter.jetty.html

File: swift-ticketing/test/swift_ticketing/fixtures.clj (hunk 3 above)
Comment (zzat) [reply swift-ticketing/test/swift_ticketing/fixtures.clj]: Added [kaocha](https://github.com/lambdaisland/kaocha) as the test runner

File: swift-ticketing/test/swift_ticketing/handlers_test.clj [hunk 4]
```diff
@@ -4,18 +4,17 @@
...
+            [clojure.walk :refer [keywordize-keys]]
             [swift-ticketing.app :refer [swift-ticketing-app]]
             [swift-ticketing.fixtures :as fixtures]
             [swift-ticketing.factory :as factory]
             [swift-ticketing.utils :as utils]
             [swift-ticketing.db.event :as db-event]
             [swift-ticketing.db.ticket :as ticket]
             [swift-ticketing.client :as client]
-            [swift-ticketing.specs :as specs]))
+            ))
```
Comment (olttwa) [inline swift-ticketing/test/swift_ticketing/handlers_test.clj:15]: Think the parentheses should be in above line as per formatting conventions

**Review by olttwa — APPROVED** [review]
one minor change. otherwise good to be merged.

**Pull request merged**